src/loader-components/_ExampleTemplate.jsx

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import styles from './ExampleTemplate.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The example component.
+   */
+  example: PropTypes.element,
+  /**
+   * The example source code.
+   */
+  exampleSrc: PropTypes.element,
+  /**
+   * The example title.
+   */
+  title: PropTypes.string,
+  /**
+   * The example description.
+   */
+  description: PropTypes.node,
+};
+
+class ExampleTemplate extends React.Component {
+  static renderHeader(title) {
+    if (title) {
+      return (
+        <div className={cx('header')}>
+          <h2 className={cx('title')}>
+            {title}
+          </h2>
+        </div>
+      );
+    }
+    return null;
+  }
+
+  static renderDescription(description) {
+    if (description) {
+      return (
+        <div className={cx('description')}>
+          {description}
+        </div>
+      );
+    }
+    return null;
+  }
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isExpanded: false,
+      isBackgroundTransparent: false,
+    };
+
+    this.handleBgToggle = this.handleBgToggle.bind(this);
+    this.handleCodeToggle = this.handleCodeToggle.bind(this);
+  }
+
+  handleBgToggle() {
+    this.setState(prevState => ({ isBackgroundTransparent: !prevState.isBackgroundTransparent }));
+  }
+
+  handleCodeToggle() {
+    this.setState(prevState => ({ isExpanded: !prevState.isExpanded }));
+  }
+
+  render() {
+    const {
+      example,
+      exampleSrc,
+      title,
+      description,
+    } = this.props;
+
+    const { isExpanded, isBackgroundTransparent } = this.state;
+
+    return (
+      <div className={cx('template')}>
+        {ExampleTemplate.renderHeader(title)}
+        <div className={cx('content', { 'dynamic-content': isBackgroundTransparent })}>
+          {description}
+          {example}
+        </div>
+        {exampleSrc
+          && (
+          <div className={cx('footer')}>
+            <div className={cx('button-container')}>
+              <button type="button" className={cx('bg-toggle')} onClick={this.handleBgToggle}>
+                Toggle Background
+              </button>
+              <button type="button" className={cx('code-toggle')} onClick={this.handleCodeToggle}>
+                <span className={cx('chevron-left')} />
+                <span>Code</span>
+                <span className={cx('chevron-right')} />
+              </button>
+            </div>
+            {isExpanded
+              && (
+              <div className={cx('code', 'is-expanded')}>
```

Comment:
you mentioned you pulled this forward from doc-template. I think this needs cleaned up a little to either always render and utilize the class styles or remove the class styles that change the height values ('code' applies height: 0, and 'is-expanded' applies height 100%)

To utilize the css classes I'd expect: 
```
<div className={cx(['code',  { 'is-expanded': isExpanded }])}>
    {exampleSrc}
<div>
```

Reply:
+1 i had intended to combine these but forgot.